### ISSUE TITLE
Support passing `MB_DB_PASS` and/or `MB_DB_USER` in combination with `MB_DB_CONNECTION_URI`

### ISSUE THREAD

**Issue by camsaul** [metabase/metabase#20122]
Related work: #19970 and #20121

IIRC `java.sql.DriverManager` has three versions of `getConnection`: 

```clj
(DriverManager/getConnection ^String connection-string)
(DriverManager/getConnection ^String connection-string ^String user ^String password)
(DriverManager/getConnection ^String connection-string ^java.util.Properties properties)
```

we use the first one for `MB_DB_CONNECTION_URI` and the last one for broken-out details. The second one would correspond perfectly to a combination of `MB_DB_CONNECTION_URI` with the username and password passed separately, so it would be pretty easy to implement. This main use cause would be if the password contained special characters that could trip up the JDBC driver when it tries to parse the connection string.

We'll need to make sure we test and document this tho.

**Comment by camsaul** [comment]
Since we're not parsing the connection string anymore and not URL-decoding stuff anymore this could work around some edge cases where you can't construct a connection string that the JDBC driver likes because of special characters

**Comment by camsaul** [comment]
Once this is implemented I think we can go ahead and close out #5750

**Comment by camsaul** [comment]
Implemented by #20135
